src/App.tsx

```diff
@@ -51,11 +52,28 @@ const App = () => {
     setModal(true)
   }
 
+  const pollNetworkType = (client: Client) => {
+    client
+      .getNetworkType()
+      .then(setNetworkType)
+      .catch(() => setNetworkType(NetworkType.UNREACHABLE))
+  }
+
   useEffect(() => {
     setApiClient(new Client(settings.nodeHost, settings.walletName, settings.password))
     saveSettings(settings)
   }, [settings])
 
+  useEffect(() => {
+    if (apiClient) {
+      const interval = setInterval(() => {
+        pollNetworkType(apiClient)
+      }, 15000)
+      pollNetworkType(apiClient)
+      return () => clearInterval(interval)
+    }
```

Comment:
I'm not sure to understand the logic here, you set an interval to poll the network every 15s, then you poll once and you directly stop the interval?

Reply:
I manually poll once because `setInterval` waits 15 seconds before its first polling. Like this we directly get the network type when launching the app. 
The returned function is a [cleanup function](https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup) which is only triggered when the component is unmounted by react. Otherwise we might have memory leaks because react will try to update a component which no longer exist. That's why we also do it when polling in `Administrate` or `SubmitVote`. Switching pages will unmount these components, so we need to return a cleanup function to stop the polling when unmounting. We have something like this in these components:
```tsx
  useEffect(() => {
    const interval = setInterval(() => {
      pollTxStatus(interval)
    }, 1000)
    pollTxStatus(interval)
    return () => clearInterval(interval)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [txResult])
```

Concerning component `App`, it is the top component in the rendering tree hierarchy, so I don't think it can be unmounted. But it is good practice to cleanup I think.

Reply:
understood, thxxx